libraries/Wire/src/utility/twi.c

```diff
@@ -373,8 +413,17 @@ void twi_stop(void)
 
   // wait for stop condition to be exectued on bus
   // TWINT is not set after a stop condition!
+  volatile uint32_t counter = twi_timeout_us/10ul; // approximate the timeout
```

Comment:
```suggestion
  // We cannot use micros() from an ISR, so approximate the timeout with cycle-counted delays
  const uint8_t us_per_loop = 10;  
  uint32_t counter = (twi_timeout_us + us_per_loop - 1)/us_per_loop; // Round up
```

No need to make this `volatile` AFAICS, there will not be an ISR interrupting this code and writing to `counter`. Also, round up, to make sure the timeout is never *shorter* than configured. And also use a constant, to prevent duplicating the 10.

Reply:
Does it make sense to use **8** as a value for **us_per_loop** variable? Is division by **8** simple to perform as it is mostly like shifting the register by three bits?

Reply:
Yup, seems like a good suggestion to me. Suggestion updated.